apps/meteor/app/lib/server/functions/getRoomByNameOrIdWithOptionToJoin.ts

```diff
@@ -40,8 +40,7 @@ export const getRoomByNameOrIdWithOptionToJoin = async ({
 			});
 		}
 
-		const rid = isObject(roomUser) ? [user._id, roomUser._id].sort().join('') : nameOrId;
-		room = await Rooms.findOneById(rid);
+		room = isObject(roomUser) ? await Rooms.findOneDirectRoomContainingAllUserIDs([...new Set([user._id, roomUser._id])]) : null;
```

Comment:
```suggestion
		room = isObject(roomUser) ? await Rooms.findOneDirectRoomContainingAllUserIDs([...new Set([user._id, roomUser._id])]) : Rooms.findOneById(rid);
```
if `roomUser` doesn't exist, then the `roomId` has been provided (and not the username)